discounts/rust/product-discounts/default/src/main.rs

```diff
@@ -3,273 +3,75 @@ use serde::{Deserialize, Serialize};
 mod api;
 use api::*;
 
-#[derive(Clone, Debug, Deserialize)]
-pub struct Payload {
-    pub input: input::Input,
-    pub configuration: Configuration,
-}
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ConfigurationValue {}
```

Comment:
`ConfigurationValue` is what's parsed from the metafield value JSON string so there's option types, and I wanted `Configuration` to actually have the types the function would use

Maybe there's a more rust-y way to do that conversion and have the result be an error otherwise?

Reply:
Oh sorry these comments make less sense without the concrete examples...